src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -935,24 +961,28 @@ public RedisSentinelConnection getSentinelConnection() {
 
 	private Jedis getActiveSentinel() {
 
-		Assert.isTrue(RedisConfiguration.isSentinelConfiguration(configuration), "SentinelConfig must not be null");
-		SentinelConfiguration sentinelConfiguration = (SentinelConfiguration) configuration;
+		Assert.isTrue(RedisConfiguration.isSentinelConfiguration(this.configuration),
+				"SentinelConfig must not be null");
+
+		SentinelConfiguration sentinelConfiguration = (SentinelConfiguration) this.configuration;
 
 		JedisClientConfig clientConfig = createSentinelClientConfig(sentinelConfiguration);
+
 		for (RedisNode node : sentinelConfiguration.getSentinels()) {
 
 			Jedis jedis = null;
 			boolean success = false;
 
 			try {
 
-				jedis = new Jedis(new HostAndPort(node.getHost(), node.getPort()), clientConfig);
+				jedis = createJedis(node, clientConfig);
+
 				if (jedis.ping().equalsIgnoreCase("pong")) {
 					success = true;
 					return jedis;
 				}
-			} catch (Exception ex) {
-				log.warn(String.format("Ping failed for sentinel host: %s", node.getHost()), ex);
+			} catch (Exception cause) {
```

Comment:
`cause` is technically wrong, please stick to `ex`

Reply:
No. It is the cause for the Exception being thrown in the block above.